secant/Features/BackupFlow/RecoveryPhraseValidation.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  RecoveryPhraseValidation.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/29/21.
+//
+
+import Foundation
+import ComposableArchitecture
+import SwiftUI
+
+typealias RecoveryPhraseValidationStore = Store<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+typealias RecoveryPhraseValidationViewStore = ViewStore<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+
+/// Represents the data of a word that has been placed into an empty position, that will be used
+/// to validate the completed phrase when all ValidationWords have been placed.
+struct ValidationWord: Equatable {
+    var groupIndex: Int
+    var word: String
+}
+
+struct RecoveryPhraseValidationState: Equatable {
+    enum Step: Equatable {
+        case initial
+        case incomplete
+        case complete
+    }
+
+    enum Route: Equatable, CaseIterable {
+        case success
+        case failure
+    }
+
+    static let wordGroupSize = 6
+    static let phraseChunks = 4
+
+    var phrase: RecoveryPhrase
+    var missingIndices: [Int]
+    var missingWordChips: [PhraseChip.Kind]
+    var validationWords: [ValidationWord]
+    var route: Route?
+    
+    var step: Step {
+        guard !validationWords.isEmpty else {
+            return  .initial
+        }
+
+        guard validationWords.count >= missingIndices.count else {
+            return .incomplete
+        }
+
+        return .complete
+    }
```

Comment:
```suggestion
    var step: Step {
        guard !validationWords.isEmpty else { return  .initial }
        guard validationWords.count >= missingIndices.count else { return .incomplete }

        return .complete
    }
```

Reply:
here's is when I realize I have a smaller monitor than yours ^^

Reply:
There are longer lines in this file already though

Reply:
I must have had a preview pinned 😬

Reply:
Ah yeah, previews will do that to you for sure.